lib/presentation/pages/intro_page.dart

```diff
@@ -19,6 +20,18 @@ class _IntroPageState extends State<IntroPage> {
   String studyType = '';
   bool showNextButton = true;
   IntroState state = IntroState.welcome;
+  bool? dndEnabled;
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      bool dnd = await FlutterDnd.getCurrentInterruptionFilter() ==
+          FlutterDnd.INTERRUPTION_FILTER_PRIORITY;
+      setState(() {
+        dnd ? dndEnabled = true : dndEnabled = false;
+      });
+    });
```

Comment:
In initState it is recommended to put the code after `super.initState();`
Also I prefer to not use `addPostFrameCallback` and instead show a loading widget if needed.

Also this code block should move to a separate function